Data-Structures/src/test/java/linkedList/LinkedListTest.java

```diff
@@ -0,0 +1,101 @@
+package linkedList;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class LinkedListTest {
+
+    // Testing for initializing empty linked list with no nodes
+    @Test
+    public void testCreateEmptyLinkedList() {
+        LinkedList test = new LinkedList();
+
+        assertNull("The head of the linked list should be null initially for empty linked list",
+                    test.head);
+    }
+
+    // Testing for inserting one single node to the linked list
+    @Test
+    public void insertOneNode() {
+        LinkedList test = new LinkedList();
+        test.insert(5);
+        Node currentNode = test.head;
+
+        assertEquals(5, currentNode.value);
+    }
+
+    // Testing for inserting multiple nodes to the linked list
+    @Test
+    public void insertMultipleNodes() {
+        LinkedList test = new LinkedList();
+        test.insert(5);
+        test.insert(7);
+        test.insert(9);
+        test.insert(3);
+        Node currentNode = test.head;
+
+        assertEquals(3, currentNode.value);
+        currentNode = currentNode.next;
+        assertEquals(9, currentNode.value);
+        currentNode = currentNode.next;
+        assertEquals(7, currentNode.value);
+        currentNode = currentNode.next;
+        assertEquals(5, currentNode.value);
+        currentNode = currentNode.next;

```

Comment:
I like this solution to figure out exactly how your list is structured inside, and it's an important thing to verify.